### PR TITLE
Transform AccountType from groups to user's attribute

### PR DESCRIPTION
**Pull request by Rotheem**
### Description

- AccountType is now a unique attribute of each user, bringing more consistency to it
- Everything that could be filtered by groups can now be filtered by account type too (endpoints, visibility, auth clients)
- Added module awareness to visibility
- Removed external filter on user creation

### Checklist

- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the documentation, if necessary

### CODE REVIEW

**Review by armanddidierjean — CHANGES_REQUESTED** [review]

File: tests/commons.py [hunk 1]
```diff
@@ -108,13 +108,13 @@ def change_redis_client_status(activated: bool) -> None:
 
 async def create_user_with_groups(
     groups: list[GroupType],
+    account_type: AccountType = AccountType.external,
```
Comment (armanddidierjean) [inline tests/commons.py:111]: I think by default we could create student users. This corresponds to the most common use case.

**Review by armanddidierjean — CHANGES_REQUESTED** [review]

File: app/utils/initialization.py [hunk 2]
```diff
@@ -81,6 +108,41 @@ def create_group_sync(
         return group
 
 
+def get_core_data_sync(schema: str, db: Session) -> models_core.CoreData | None:
```
[the remark this answers is not in the document]
Comment (Rotheem) [reply app/utils/initialization.py:111]: It is already the same

**Review by armanddidierjean — CHANGES_REQUESTED** [review]
Could you increase test coverage in `endpoints_user` and `endpoints_core`?

**Review by foucblg — APPROVED** [review]

File: app/utils/initialization.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rotheem) [reply app/utils/initialization.py]: Done

**Review by Daihecyy — APPROVED** [review]
Consider changing list to generator in tools.py to improve drastically performances

File: app/core/groups/groups_type.py [hunk 3]
```diff
@@ -12,12 +12,12 @@ class GroupType(str, Enum):
...
-    staff = "703056c4-be9d-475c-aa51-b7fc62a96aaa"
-    association = "29751438-103c-42f2-b09b-33fbb20758a7"
-    external = "b1cd979e-ecc1-4bd0-bc2b-4dad2ba8cded"
-    demo = "ae4d1866-e7d9-4d7f-bee7-e0dda24d8dd8"
+    # student = "39691052-2ae5-4e12-99d0-7a9f5f2b0136"
+    # former_student = "ab4c7503-41b3-11ee-8177-089798f1a4a5"
+    # staff = "703056c4-be9d-475c-aa51-b7fc62a96aaa"
+    # association = "29751438-103c-42f2-b09b-33fbb20758a7"
+    # external = "b1cd979e-ecc1-4bd0-bc2b-4dad2ba8cded"
+    # demo = "ae4d1866-e7d9-4d7f-bee7-e0dda24d8dd8"
```
Comment (armanddidierjean) [inline app/core/groups/groups_type.py:20]: I believe we can delete these lines

File: app/dependencies.py [hunk 4]
```diff
@@ -286,24 +286,70 @@ async def get_current_user(
...
+    To check if the user is not external and is the member of a group, use is_user_in generator
+    To check if the user has an ecl account type, use is_user_an_ecl_member dependency
     """
-    return user
+
+    excluded_groups = excluded_groups or []
+    excluded_account_types = excluded_account_types or []
+    included_account_types = included_account_types or list(AccountType)
+
+    def is_user(
```
Comment (armanddidierjean) [inline app/dependencies.py:309]: Three cases remains untested, would you mind adding a test for them?
- `excluded_account_types`
- `is_user_external`
- `included_account_types`

**Review by armanddidierjean — APPROVED** [review]
Really impressive and brilliant PR!

**Comment by Rotheem** [comment]
> Consider changing list to generator in tools.py to improve drastically performances

Do not consider this comment it's BS

**Pull request merged**